controllers/dnspolicy_controller_single_cluster_test.go

```diff
@@ -71,6 +71,12 @@ var _ = Describe("DNSPolicy Single Cluster", func() {
 				},
 			}
 			gateway.Status.Listeners = []gatewayapiv1.ListenerStatus{
+				{
+					Name:           "foo",
+					SupportedKinds: []gatewayapiv1.RouteGroupKind{},
+					AttachedRoutes: 0,
+					Conditions:     []metav1.Condition{},
+				},
```

Comment:
Just adding the status won't replicate the issue, the code you modified is inside an iteration of the spec listeners so you will also need to add it there so it goes into the check for attached routes and returns when it sees it's 0, see the example [here](https://github.com/Kuadrant/kuadrant-operator/pull/655#issuecomment-2120383882)  

```.go
gateway = NewGatewayBuilder(TestGatewayName, gatewayClass.Name, testNamespace).
	WithHTTPListener("foo", "foo.example.com"). //<-- Listener with no attached routes
	WithHTTPListener(TestListenerNameOne, TestHostOne).
	WithHTTPListener(TestListenerNameWildcard, TestHostWildcard).
	Gateway
```

Reply:
Done. Also added a check for the enforced condition on the policy since this was the original cause for creating the issue